flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/lookup/AsyncLookupJoinRunner.java

```diff
@@ -141,13 +166,22 @@ public void close() throws Exception {
                 rf.close();
             }
         }
+        if (cacheHandler != null) {
+            cacheHandler.getCache().close();
```

Comment:
Should unregister the cache as well. Otherwise, memory leaks. That means `LookupCacheManager` may need to track the reference count.